src/AutoROM.py

```diff
@@ -160,44 +160,35 @@ def torrent_tar_to_buffer():
 
-    tries: int = 0
-    success: bool = False
-    while success is False:
-        # libtorrent params
-        ses = lt.session()
-        params = lt.parse_magnet_uri(uri)
-        params.save_path = save_path
-        handle: lt.torrent_handle = ses.add_torrent(params)
-
-        # download roms as long as state is not seeding
-        timeit = 1
-        while handle.status().state not in {4, 5}:
+    # libtorrent params
+    ses = lt.session()
+    params = lt.parse_magnet_uri(uri)
+    params.save_path = save_path
+    handle: lt.torrent_handle = ses.add_torrent(params)
+
+    # download roms as long as state is not seeding
+    timeit = 0
+    while handle.status().state not in {4, 5}:
+        if timeit >= 180:
+            raise RuntimeError("Terminating attempt to download ROMs after 180 seconds, this has failed, please report it.")
+        elif timeit % 5 == 0:
             status: lt.torrent_status = handle.status()
-
-            # some sleep helps
-            time.sleep(1.0)
-            timeit += 1
-
-            if timeit >= 60:
-                print("Terminating attempt to download ROMs after 60 seconds, trying again", file=sys.stderr)
-                break
-            elif timeit % 5 == 0:
-                print(f"time={timeit / 5:.1f}/60 seconds - Trying to download atari roms\n"
-                      f"\tcurrent status={status_meaning.get(status.state, 'unknown')} ({status.state})\n"
-                      f"\ttotal downloaded bytes={status.total_download}\n"
-                      f"\ttotal payload download={status.total_payload_download}\n"
-                      f"\ttotal failed bytes={status.total_failed_bytes}",
-                      file=sys.stderr)
-
-        success = handle.status().state in {4, 5}
-        tries += 1
-
-        # seed for 20 seconds to help the network
-        if handle.status().state in {4, 5}:
-            time.sleep(20.0)
-
-        if tries >= 3:
-            raise RuntimeError("Tried to download ROMs 3 times, which have all failed, please try again or report this issue.")
+            print(f"time={timeit}/180 seconds - Trying to download atari roms\n"
+                  f"\tcurrent status={status_meaning.get(status.state, 'unknown')} ({status.state})\n"
+                  f"\ttotal downloaded bytes={status.total_download}\n"
+                  f"\ttotal payload download={status.total_payload_download}\n"
+                  f"\ttotal failed bytes={status.total_failed_bytes}",
+                  file=sys.stderr)
+
+        # some sleep helps
+        time.sleep(1.0)
+        timeit += 1
+
+    # seed for 20 seconds to help the network
+    if handle.status().state in {4, 5}:
+        print("Download complete, seeding for 20 seconds to assist torrent network.")
+        time.sleep(20.0)
```

Comment:
Would we want
```python
for _ in tqdm(range(20)):
    time.sleep(1.0)
```

Reply:
Actually, probably not, we can save one dependency from being required. I'll make the relevant changes in the setup and requirements.txt.